rustc-1.29.0-src.patch

```diff
@@ -83,3 +83,16 @@
 +pub fn check_for(x: Feature) -> bool {
 +    false
 +}
+
+# This test doesn't actually detect x32 properly
```

Comment:
It actually does detect x32 properly, I think: unlike other environments, Rust requires target names to match its expectations exactly. While `x86_64-linux-gnux32` may be a valid triple in most environments, it is not a valid Rust triple. For most targets where mrustc uses a different triple name compared to what Rust uses, it works out okay because nothing looks at the exact triple name too closely, but if it doesn't work out here, it may be better to just match expectations and change the triple name.

Aside from that: I have adapted this PR to mrustc 0.9 except with the Rust-standard triple name, applied the fix for #167, quickly hacked together a fix for #168 (that I should clean up, reapply to the main branch, and submit as a PR), and with that mrustc for x32 seems to be in a very good state: it was able to successfully build rustc 1.29.0 for x32.

Disclaimer: building rustc for x32 requires patching even when mrustc isn't part of the picture; I had applied those patches to the version in mrustc as well to get a fair comparison. None of those patches were to the mrust parts of the code.

Reply:
Agree on the quad name, I must've been confused when writing that.

Did you arrive at something like [this](https://github.com/thepowersgang/mrustc/compare/master...nabijaczleweli:x32-v2)? It builds and the strings embedded in rustc look right, but I'd like confirmation from someone who has actually managed to use this before before I update the PR.

Reply:
Yes, that looks like it's pretty much what I was using, I had a few minor differences, but it was basically just code style, nothing that should affect results. The `--no-show-signature` thing is unrelated to x32, but yes I have the same Git config so I needed that too. It might be good to split that into a separate PR.

Reply:
Great, thanks!